Description: Enablement/disablement via /etc/default/waagent
 Allow users to control the startup of waagent via /etc/default/waagent.
 Setting "WAAGENT_ENABLED=0" will turn off the agent at boot time.
Author: Ben Howard <ben.howard@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1014864

--- a/waagent
+++ b/waagent
@@ -1979,6 +1979,13 @@ author "Ben Howard <ben.howard@canonical
 start on (filesystem and started rsyslog)
 
 pre-start script
+
+	[ -r /etc/default/walinuxagent ] && . /etc/default/walinuxagent
+
+	if [ "$WALINUXAGENT_ENABLED" != "1" ]; then
+		exit 1
+	fi
+
 	if [ ! -x /usr/sbin/waagent ]; then
 		exit 1
 	fi
